HungryWolfs/DetailsMeal/DetailViewController.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  DetailViewController.swift
+//  HungryWolfs
+//
+//  Created by Wolfpack Digital on 18.08.2021.
+//
+
+import UIKit
+import Kingfisher
+
+class DetailViewController: UIViewController
+{
+    
+    @IBOutlet weak var mealNameLabel: UILabel!
+    @IBOutlet weak var mealImage: UIImageView!
+    @IBOutlet weak var measure1Label: UILabel!
+    @IBOutlet weak var measure2Label: UILabel!
+    @IBOutlet weak var measure3Label: UILabel!
+    @IBOutlet weak var ingredient1Label: UILabel!
+    @IBOutlet weak var ingredient2Label: UILabel!
+    @IBOutlet weak var ingredient3Label: UILabel!
+    @IBOutlet weak var mealTag1Label: UILabel!
+    @IBOutlet weak var mealTag2Label: UILabel!
+    @IBOutlet weak var mealTag3Label: UILabel!
+    @IBOutlet weak var mealDescription: UILabel!
+    @IBOutlet weak var checkButton1: UIButton!
+    @IBOutlet weak var checkButton2: UIButton!
+    @IBOutlet weak var checkButton3: UIButton!
+    @IBOutlet weak var tagView1: UIView!
+    @IBOutlet weak var tagView2: UIView!
+    @IBOutlet weak var tagView3: UIView!
+    @IBOutlet weak var favouriteButton: UIButton!
+    
+    var id = ""
+    
+    var favId = ""
+    var favName = ""
+    var favImg = ""
+    
+    private let viewModelDetails: DetailsViewModel = DetailsViewModel()
+   
+    override func viewDidLoad()
+    {
+        super.viewDidLoad()
+        
+        // butoanele bifabile
+        checkButton1.setImage(UIImage(systemName: "checkmark.square"),
+                          for: [.selected])
+        checkButton2.setImage(UIImage(systemName: "checkmark.square"),
+                          for: [.selected])
+        checkButton3.setImage(UIImage(systemName: "checkmark.square"),
+                          for: [.selected])
+        
+    
+        viewModelDetails.getDetails(id: id, completion: { [weak self] in
+            
+            // pentru favourites
```

Comment:
You can use a "guard let self = self else { return }" here at the top, so you don't have to use "self?" everywhere below.